src/shared/install.c

```diff
@@ -1413,6 +1416,38 @@ static int unit_file_search(
 
         assert(info->name);
 
+        if (unit_name_is_valid(info->name, UNIT_NAME_INSTANCE)) {
+                r = unit_name_template(info->name, &template);
+                if (r < 0)
+                        return r;
+        }
+
+        STRV_FOREACH(p, paths->search_path) {
+                _cleanup_free_ char *path = NULL;
+
+                path = strjoin(*p, "/", info->name, ".d", NULL);
```

Comment:
`path_join` should be used in general, it is needed for `arg_root` support. So you probably need to concatentate `info->name` and `".d"` first using strjoina (on a seperate line), outside of the loop, and then call `path_join` here.